Userland/Libraries/LibCore/System.cpp

```diff
@@ -997,6 +997,16 @@ ErrorOr<int> mkstemp(Span<char> pattern)
     return fd;
 }
 
+ErrorOr<char*> mkdtemp(Span<char> pattern)
+{
+    auto* path = ::mkdtemp(pattern.data());
+    if (path == nullptr) {
+        return Error::from_string_literal("Failed to create temporary directory!");
```

Comment:
We can propagate the actual error: `return Error::from_errno(errno);`

see https://github.com/SerenityOS/serenity/blob/1dc074fc18970915c08b0b9b5e07945c67bda7e4/Userland/Libraries/LibC/stdlib.cpp#L766-L769